File: ESHOP/src/main/java/com/example/teke/ESHOP/model/Customer.java
package com.example.teke.ESHOP.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.Data;


import jakarta.persistence.*;

import java.util.List;
import java.util.UUID;


@JsonPropertyOrder({
        "id",
        "username",
        "password",
        "name",
        "surname",
        "email",
        "phone",
        "address",
        "agreement",
        "activeOrder"
})
@Data
@Entity
public class Customer {



    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;
    private String username;
    private String password;
    private String name;
    private String surname;
    private String email;
    private String phone;
    private String address;
    private Boolean agreement;
    private Boolean activeOrder = false;
    @Enumerated(EnumType.STRING)
    private Role role ;

    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
    private List<UserInteraction> interactions;

}

File: eshop-ui/src/app/models/customer.ts
export class Customer {
    id :number;
    username :string;
    password :string;
    name :string;
    surname :string;
    email :string;
    phone :string;
    address :string;
    agreement :Boolean;
    activeOrder :Boolean;

    constructor(username: string = '', password: string = '', name: string = '', surname: string = '', email: string = '', phone: string = '', address: string = '', agreement: boolean = false, activeOrder: boolean = false) {
        this.username = username;
        this.password = password;
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.agreement = agreement;
        this.activeOrder = activeOrder;
    }
    // Giriş yaparken kullanılacak constructor
    static login(username: string, password: string): Customer {
        return new Customer(username, password);
    }

}
